src/main/haskell/kore/src/Kore/ASTUtils/AlphaCompare.hs

```diff
@@ -36,133 +38,380 @@ import qualified Kore.Domain.Builtin as Domain
+            (EqualsPattern eq1, EqualsPattern eq2) -> compareEquals eq1 eq2
+            (ExistsPattern ex1, ExistsPattern ex2) -> compareExists ex1 ex2
+            (ForallPattern fa1, ForallPattern fa2) -> compareForall fa1 fa2
+            (FloorPattern flr1, FloorPattern flr2) -> compareFloor flr1 flr2
+            (IffPattern iff1, IffPattern iff2) -> compareIff iff1 iff2
+            (ImpliesPattern imp1, ImpliesPattern imp2) ->
+                compareImplies imp1 imp2
+            (InPattern in1, InPattern in2) -> compareIn in1 in2
+            (NextPattern nxt1, NextPattern nxt2) -> compareNext nxt1 nxt2
+            (NotPattern not1, NotPattern not2) -> compareNot not1 not2
+            (OrPattern or1, OrPattern or2) -> compareOr or1 or2
+            (RewritesPattern rew1, RewritesPattern rew2) ->
+                compareRewrites rew1 rew2
+            (StringLiteralPattern lit1, StringLiteralPattern lit2) ->
+                compareStringLiteral lit1 lit2
+            (TopPattern top1, TopPattern top2) -> compareTop top1 top2
+            (VariablePattern var1, VariablePattern var2) ->
+                compareVariable var1 var2
+            _ -> return False
+
+    compareChildren children1 children2 =
+        and <$> Reader.zipWithM alphaEqWorker children1 children2
+
+    compareAnd
+        And
+            { andSort = andSort1
+            , andFirst = andFirst1
+            , andSecond = andSecond1
+            }
+        And
+            { andSort = andSort2
+            , andFirst = andFirst2
+            , andSecond = andSecond2
+            }
+      | andSort1 == andSort2 =
+        (&&)
+            <$> alphaEqWorker andFirst1 andFirst2
+            <*> alphaEqWorker andSecond1 andSecond2
+      | otherwise = return False
+
+    compareBottom
+        Bottom { bottomSort = bottomSort1 }
+        Bottom { bottomSort = bottomSort2 }
+      =
+        return (bottomSort1 == bottomSort2)
+
+    compareCeil
+        Ceil
+            { ceilOperandSort = ceilOperandSort1
+            , ceilResultSort = ceilResultSort1
+            , ceilChild = ceilChild1
+            }
+        Ceil
+            { ceilOperandSort = ceilOperandSort2
+            , ceilResultSort = ceilResultSort2
+            , ceilChild = ceilChild2
+            }
+      | sameSort = alphaEqWorker ceilChild1 ceilChild2
+      | otherwise = return False
+      where
+        sameSort =
+            ceilOperandSort1 == ceilOperandSort2
+            && ceilResultSort1 == ceilResultSort2
+
+    compareEquals
+        Equals
+            { equalsOperandSort = equalsOperandSort1
+            , equalsResultSort = equalsResultSort1
+            , equalsFirst = equalsFirst1
+            , equalsSecond = equalsSecond1
+            }
+        Equals
+            { equalsOperandSort = equalsOperandSort2
+            , equalsResultSort = equalsResultSort2
+            , equalsFirst = equalsFirst2
+            , equalsSecond = equalsSecond2
```

Comment:
I guess I can't object since it still works, but, is there a reason this file becomes so much longer..?

Reply:
It doesn't use the pattern synonyms in `Kore.ASTUtils.SmartPatterns`, which are weaker now.